lib/date_range.rb

```diff
@@ -0,0 +1,57 @@
+require 'date'
+
+module Hotel
+  class DateRange
+    attr_reader :start_date, :end_date
+
+    def initialize(start_date, end_date)
+      @start_date = Date.parse(start_date)
+      @end_date = Date.parse(end_date)
+    end
+
+    def check_valid_dates
+
+      raise StandardError, "Invalid dates entered" if @start_date == nil || @end_date == nil
+
+      raise StandardError, "End date cannot be before start date" if (@end_date - @start_date) <= 0
```

Comment:
Instead of raising a `StandardError`, best practice is to define your own appropriately named exception type that inherits from `StandardError` and raise that. The reason is that `StandardError` is very broad, and if the code that calls this method has to `rescue StandardError`, that will catch things like `ArgumentError`s and `NameError`s that indicate true bugs.